indiapi.h

```diff
@@ -495,3 +497,9 @@ typedef struct _IBLOBVectorProperty /* BLOB vector property descriptor */
  * with actual array, not pointer.
  */
 #define NARRAY(a) (sizeof(a) / sizeof(a[0]))
+
+/**
+ * @brief Bails out if memory pointer is 0. Prints file and function.
+ */
+#define assert_mem(p) if((p) == 0) { fprintf(stderr, "%s(%s): Failed to allocate memory\n", __FILE__, __func__); exit(1); }
```

Comment:
Oh sorry, by bailing out I meant bailing out from the function, not from the program!

Reply:
This is probably OK actually as it doesn't change the behavior, except when it avoids corruption. At some point the memory shortage should be managed by freeing resources that can be freed and retrying. Future!

Reply:
One thing you can do is `assert(0); return;` depending on the prototype of the function you add your `assert_mem` into. With this, debug code asserts, and production code returns to the caller without damaging the memory.

Reply:
Thanks, I agree. Its better to die hard so the user have a chance to restart instead of possibly wasting a lot of time on a system that behaves erratic.

Reply:
let me know once this is ready.

Reply:
As far as I know it should be good now

Reply:
Oh, I missed @TallFurrMan last comment.
I still think it's better to exit here since the code is not really written to handle memory allocation in general and would require quite a bit of coding. Standing in the middle of the night fotographing I would rather do a clean restart instead of having a half broken exposure sequence running.

Reply:
Oh, I missed @TallFurrMan last comment.
I still think it's better to exit here since the code is not really written to handle memory allocation in general and would require quite a bit of coding. Standing in the middle of the night fotographing I would rather do a clean restart instead of having a half broken exposure sequence running.